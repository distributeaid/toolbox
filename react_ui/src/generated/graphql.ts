/* eslint-disable */
// This file was automatically generated and should not be edited.
// --
import gql from 'graphql-tag';
import * as ApolloReactCommon from '@apollo/client';
import * as ApolloReactHooks from '@apollo/client';
export type Maybe<T> = T | null;
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};

export type Group = {
  readonly __typename?: 'Group';
  readonly description: Maybe<Scalars['String']>;
  readonly donationForm: Maybe<Scalars['String']>;
  readonly donationFormResults: Maybe<Scalars['String']>;
  readonly donationLink: Maybe<Scalars['String']>;
  readonly id: Scalars['ID'];
  readonly name: Scalars['String'];
  readonly requestForm: Maybe<Scalars['String']>;
  readonly requestFormResults: Maybe<Scalars['String']>;
  readonly slackChannelName: Maybe<Scalars['String']>;
  readonly slug: Scalars['String'];
  readonly type: GroupType;
  readonly volunteerForm: Maybe<Scalars['String']>;
  readonly volunteerFormResults: Maybe<Scalars['String']>;
};

export type GroupInput = {
  readonly description: Maybe<Scalars['String']>;
  readonly donationForm: Maybe<Scalars['String']>;
  readonly donationFormResults: Maybe<Scalars['String']>;
  readonly donationLink: Maybe<Scalars['String']>;
  readonly name: Maybe<Scalars['String']>;
  readonly requestForm: Maybe<Scalars['String']>;
  readonly requestFormResults: Maybe<Scalars['String']>;
  readonly slackChannelName: Maybe<Scalars['String']>;
  readonly slug: Maybe<Scalars['String']>;
  readonly volunteerForm: Maybe<Scalars['String']>;
  readonly volunteerFormResults: Maybe<Scalars['String']>;
};

export enum GroupType {
  M4D_CHAPTER = 'M4D_CHAPTER'
}

export type RootMutationType = {
  readonly __typename?: 'RootMutationType';
  readonly createGroup: Maybe<Group>;
  readonly deleteGroup: Maybe<Group>;
  readonly updateGroup: Maybe<Group>;
};


export type RootMutationTypeCreateGroupArgs = {
  groupInput: GroupInput;
};


export type RootMutationTypeDeleteGroupArgs = {
  id: Scalars['ID'];
};


export type RootMutationTypeUpdateGroupArgs = {
  groupInput: GroupInput;
  id: Scalars['ID'];
};

export type RootQueryType = {
  readonly __typename?: 'RootQueryType';
  readonly countGroups: Maybe<Scalars['Int']>;
  readonly group: Maybe<Group>;
  readonly groupBySlug: Maybe<Group>;
  readonly groups: Maybe<ReadonlyArray<Maybe<Group>>>;
  readonly healthCheck: Maybe<Scalars['String']>;
  readonly session: Maybe<Session>;
};


export type RootQueryTypeGroupArgs = {
  id: Scalars['ID'];
};


export type RootQueryTypeGroupBySlugArgs = {
  slug: Scalars['String'];
};

export type Session = {
  readonly __typename?: 'Session';
  readonly id: Maybe<Scalars['ID']>;
  readonly userId: Maybe<Scalars['ID']>;
};

export type CreateChapterMutationVariables = {
  groupInput: GroupInput;
};


export type CreateChapterMutation = { readonly __typename?: 'RootMutationType', readonly createGroup: Maybe<{ readonly __typename?: 'Group', readonly id: string, readonly slug: string }> };

export type GetChapterQueryVariables = {
  id: Scalars['ID'];
};


export type GetChapterQuery = { readonly __typename?: 'RootQueryType', readonly group: Maybe<{ readonly __typename?: 'Group', readonly id: string, readonly name: string, readonly slug: string, readonly description: Maybe<string>, readonly donationForm: Maybe<string>, readonly donationLink: Maybe<string>, readonly donationFormResults: Maybe<string>, readonly slackChannelName: Maybe<string>, readonly volunteerForm: Maybe<string>, readonly volunteerFormResults: Maybe<string>, readonly requestForm: Maybe<string>, readonly requestFormResults: Maybe<string> }> };

export type GetChapterBySlugQueryVariables = {
  slug: Scalars['String'];
};


export type GetChapterBySlugQuery = { readonly __typename?: 'RootQueryType', readonly groupBySlug: Maybe<{ readonly __typename?: 'Group', readonly id: string, readonly name: string, readonly slug: string, readonly description: Maybe<string>, readonly donationForm: Maybe<string>, readonly donationLink: Maybe<string>, readonly donationFormResults: Maybe<string>, readonly slackChannelName: Maybe<string>, readonly volunteerForm: Maybe<string>, readonly volunteerFormResults: Maybe<string>, readonly requestForm: Maybe<string>, readonly requestFormResults: Maybe<string> }> };

export type GetChapterListQueryVariables = {};


export type GetChapterListQuery = { readonly __typename?: 'RootQueryType', readonly groups: Maybe<ReadonlyArray<Maybe<{ readonly __typename?: 'Group', readonly id: string, readonly slug: string, readonly description: Maybe<string>, readonly name: string }>>> };

export type UpdateChapterMutationVariables = {
  id: Scalars['ID'];
  groupInput: GroupInput;
};


export type UpdateChapterMutation = { readonly __typename?: 'RootMutationType', readonly updateGroup: Maybe<{ readonly __typename?: 'Group', readonly id: string, readonly slug: string }> };


export const CreateChapterDocument = gql`
    mutation createChapter($groupInput: GroupInput!) {
  createGroup(groupInput: $groupInput) {
    id
    slug
  }
}
    `;
export type CreateChapterMutationFn = ApolloReactCommon.MutationFunction<CreateChapterMutation, CreateChapterMutationVariables>;

/**
 * __useCreateChapterMutation__
 *
 * To run a mutation, you first call `useCreateChapterMutation` within a React component and pass it any options that fit your needs.
 * When your component renders, `useCreateChapterMutation` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [createChapterMutation, { data, loading, error }] = useCreateChapterMutation({
 *   variables: {
 *      groupInput: // value for 'groupInput'
 *   },
 * });
 */
export function useCreateChapterMutation(baseOptions?: ApolloReactHooks.MutationHookOptions<CreateChapterMutation, CreateChapterMutationVariables>) {
        return ApolloReactHooks.useMutation<CreateChapterMutation, CreateChapterMutationVariables>(CreateChapterDocument, baseOptions);
      }
export type CreateChapterMutationHookResult = ReturnType<typeof useCreateChapterMutation>;
export type CreateChapterMutationResult = ApolloReactCommon.MutationResult<CreateChapterMutation>;
export type CreateChapterMutationOptions = ApolloReactCommon.BaseMutationOptions<CreateChapterMutation, CreateChapterMutationVariables>;
export const GetChapterDocument = gql`
    query getChapter($id: ID!) {
  group(id: $id) {
    id
    name
    slug
    description
    donationForm
    donationLink
    donationFormResults
    slackChannelName
    volunteerForm
    volunteerFormResults
    requestForm
    requestFormResults
  }
}
    `;

/**
 * __useGetChapterQuery__
 *
 * To run a query within a React component, call `useGetChapterQuery` and pass it any options that fit your needs.
 * When your component renders, `useGetChapterQuery` returns an object from Apollo Client that contains loading, error, and data properties 
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useGetChapterQuery({
 *   variables: {
 *      id: // value for 'id'
 *   },
 * });
 */
export function useGetChapterQuery(baseOptions?: ApolloReactHooks.QueryHookOptions<GetChapterQuery, GetChapterQueryVariables>) {
        return ApolloReactHooks.useQuery<GetChapterQuery, GetChapterQueryVariables>(GetChapterDocument, baseOptions);
      }
export function useGetChapterLazyQuery(baseOptions?: ApolloReactHooks.LazyQueryHookOptions<GetChapterQuery, GetChapterQueryVariables>) {
          return ApolloReactHooks.useLazyQuery<GetChapterQuery, GetChapterQueryVariables>(GetChapterDocument, baseOptions);
        }
export type GetChapterQueryHookResult = ReturnType<typeof useGetChapterQuery>;
export type GetChapterLazyQueryHookResult = ReturnType<typeof useGetChapterLazyQuery>;
export type GetChapterQueryResult = ApolloReactCommon.QueryResult<GetChapterQuery, GetChapterQueryVariables>;
export const GetChapterBySlugDocument = gql`
    query getChapterBySlug($slug: String!) {
  groupBySlug(slug: $slug) {
    id
    name
    slug
    description
    donationForm
    donationLink
    donationFormResults
    slackChannelName
    volunteerForm
    volunteerFormResults
    requestForm
    requestFormResults
  }
}
    `;

/**
 * __useGetChapterBySlugQuery__
 *
 * To run a query within a React component, call `useGetChapterBySlugQuery` and pass it any options that fit your needs.
 * When your component renders, `useGetChapterBySlugQuery` returns an object from Apollo Client that contains loading, error, and data properties 
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useGetChapterBySlugQuery({
 *   variables: {
 *      slug: // value for 'slug'
 *   },
 * });
 */
export function useGetChapterBySlugQuery(baseOptions?: ApolloReactHooks.QueryHookOptions<GetChapterBySlugQuery, GetChapterBySlugQueryVariables>) {
        return ApolloReactHooks.useQuery<GetChapterBySlugQuery, GetChapterBySlugQueryVariables>(GetChapterBySlugDocument, baseOptions);
      }
export function useGetChapterBySlugLazyQuery(baseOptions?: ApolloReactHooks.LazyQueryHookOptions<GetChapterBySlugQuery, GetChapterBySlugQueryVariables>) {
          return ApolloReactHooks.useLazyQuery<GetChapterBySlugQuery, GetChapterBySlugQueryVariables>(GetChapterBySlugDocument, baseOptions);
        }
export type GetChapterBySlugQueryHookResult = ReturnType<typeof useGetChapterBySlugQuery>;
export type GetChapterBySlugLazyQueryHookResult = ReturnType<typeof useGetChapterBySlugLazyQuery>;
export type GetChapterBySlugQueryResult = ApolloReactCommon.QueryResult<GetChapterBySlugQuery, GetChapterBySlugQueryVariables>;
export const GetChapterListDocument = gql`
    query getChapterList {
  groups {
    id
    slug
    description
    name
  }
}
    `;

/**
 * __useGetChapterListQuery__
 *
 * To run a query within a React component, call `useGetChapterListQuery` and pass it any options that fit your needs.
 * When your component renders, `useGetChapterListQuery` returns an object from Apollo Client that contains loading, error, and data properties 
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useGetChapterListQuery({
 *   variables: {
 *   },
 * });
 */
export function useGetChapterListQuery(baseOptions?: ApolloReactHooks.QueryHookOptions<GetChapterListQuery, GetChapterListQueryVariables>) {
        return ApolloReactHooks.useQuery<GetChapterListQuery, GetChapterListQueryVariables>(GetChapterListDocument, baseOptions);
      }
export function useGetChapterListLazyQuery(baseOptions?: ApolloReactHooks.LazyQueryHookOptions<GetChapterListQuery, GetChapterListQueryVariables>) {
          return ApolloReactHooks.useLazyQuery<GetChapterListQuery, GetChapterListQueryVariables>(GetChapterListDocument, baseOptions);
        }
export type GetChapterListQueryHookResult = ReturnType<typeof useGetChapterListQuery>;
export type GetChapterListLazyQueryHookResult = ReturnType<typeof useGetChapterListLazyQuery>;
export type GetChapterListQueryResult = ApolloReactCommon.QueryResult<GetChapterListQuery, GetChapterListQueryVariables>;
export const UpdateChapterDocument = gql`
    mutation updateChapter($id: ID!, $groupInput: GroupInput!) {
  updateGroup(id: $id, groupInput: $groupInput) {
    id
    slug
  }
}
    `;
export type UpdateChapterMutationFn = ApolloReactCommon.MutationFunction<UpdateChapterMutation, UpdateChapterMutationVariables>;

/**
 * __useUpdateChapterMutation__
 *
 * To run a mutation, you first call `useUpdateChapterMutation` within a React component and pass it any options that fit your needs.
 * When your component renders, `useUpdateChapterMutation` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [updateChapterMutation, { data, loading, error }] = useUpdateChapterMutation({
 *   variables: {
 *      id: // value for 'id'
 *      groupInput: // value for 'groupInput'
 *   },
 * });
 */
export function useUpdateChapterMutation(baseOptions?: ApolloReactHooks.MutationHookOptions<UpdateChapterMutation, UpdateChapterMutationVariables>) {
        return ApolloReactHooks.useMutation<UpdateChapterMutation, UpdateChapterMutationVariables>(UpdateChapterDocument, baseOptions);
      }
export type UpdateChapterMutationHookResult = ReturnType<typeof useUpdateChapterMutation>;
export type UpdateChapterMutationResult = ApolloReactCommon.MutationResult<UpdateChapterMutation>;
export type UpdateChapterMutationOptions = ApolloReactCommon.BaseMutationOptions<UpdateChapterMutation, UpdateChapterMutationVariables>;


