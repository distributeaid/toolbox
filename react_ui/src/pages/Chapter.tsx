import React from 'react'
import { Helmet } from 'react-helmet'
import { useTranslation } from 'react-i18next'

import { Background } from '../components/Background'
import { BackLink } from '../components/BackLink'
import { BorderBlock } from '../components/BorderBlock'
import { ContentContainer } from '../components/ContentContainer'
import { FAQFooter } from '../components/FAQFooter'
import { MainContent } from '../components/MainContent'
import { MainHeader } from '../components/MainHeader'
import { P } from '../components/P'
import { PreHeader } from '../components/PreHeader'
import { ShadowButtonLink } from '../components/ShadowButtonLink'
import { useGetChapterBySlugQuery } from '../generated/graphql'
import { countryCodeToName, usStateCodeToName } from '../util/placeCodeToName'

type Props = {
  slug: string
}

export const Chapter: React.FC<Props> = ({ slug }) => {
  const { t } = useTranslation()
  const { loading, error, data } = useGetChapterBySlugQuery({
    variables: { slug },
  })

  const chapter = loading || error ? undefined : data?.groupBySlug

  return (
    <MainContent>
      <Background />

      <ContentContainer>
        <div className="pb-16 sm:pb-32">
          <BackLink to="/chapters">{t('chapter.backLink')}</BackLink>
        </div>

        {loading && <div>Loading...</div>}
        {error && <PreHeader>Chapter "{slug}" not found</PreHeader>}
        {chapter && (
          <>
            <Helmet>
              <meta
                property="og:title"
                content={`Masks For Docs – ${chapter.name}`}></meta>
            </Helmet>
            <PreHeader>
              {chapter.location.countryCode === 'US'
                ? usStateCodeToName(chapter.location.province)
                : chapter.location.province}{' '}
              — {countryCodeToName(chapter.location.countryCode)}
            </PreHeader>

            <MainHeader>{chapter.name}</MainHeader>
            {chapter.leader && (
              <div className="mb-8">
                <div className="font-body text-gray-500 text-xs">
                  {t('chapter.leadHeader')}
                </div>
                <div className="font-body text-lg font-bold">
                  {chapter.leader}
                </div>
              </div>
            )}
          </>
        )}
      </ContentContainer>

      {chapter && (
        <BorderBlock>
          <ContentContainer>
            <div className="sm:flex py-6 sm:py-12">
              <P
                className="sm:w-7/12 pb-6 whitespace-pre-line"
                data-testid="description">
                {chapter.description}
              </P>

              <div className="sm:w-5/12 sm:pr-2 z-30">
                {chapter.requestForm && (
                  <ShadowButtonLink
                    external={true}
                    className="bg-blue-600 w-full mb-8"
                    to={chapter.requestForm}>
                    {t('chapter.requestSuppliesLink')}
                  </ShadowButtonLink>
                )}

                {chapter.donationForm && (
                  <ShadowButtonLink
                    external={true}
                    className="bg-red-600 w-full mb-8"
                    to={chapter.donationForm}>
                    {t('chapter.donateSuppliesLink')}
                  </ShadowButtonLink>
                )}

                {chapter.volunteerForm && (
                  <ShadowButtonLink
                    external={true}
                    className="bg-purple-600 w-full"
                    to={chapter.volunteerForm}>
                    {t('chapter.volunteerLink')}
                  </ShadowButtonLink>
                )}
              </div>
            </div>
          </ContentContainer>
        </BorderBlock>
      )}

      <ContentContainer>
        <FAQFooter />
      </ContentContainer>
    </MainContent>
  )
}
