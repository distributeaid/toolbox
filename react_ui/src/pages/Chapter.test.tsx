import { render } from '@testing-library/react'
import React from 'react'
import { MemoryRouter } from 'react-router-dom'

import { Group, useGetChapterBySlugQuery } from '../generated/graphql'
import { Chapter } from './Chapter'

jest.mock('../generated/graphql', () => ({
  useGetChapterBySlugQuery: jest.fn(),
}))

const renderUI = (slug: string) =>
  render(
    <MemoryRouter>
      <Chapter slug={slug} />
    </MemoryRouter>
  )

it('loads the chapter details and renders the page', () => {
  ;(useGetChapterBySlugQuery as jest.Mock).mockReturnValueOnce({
    loading: false,
    data: {
      groupBySlug: {
        id: '1',
        slug: 'oakland',
        name: 'Oakland',
        description: 'the Oakland group',
        donationForm: 'donation form link',
        donationFormResults: 'donation form results link',
        donationLink: 'donation link',
        volunteerForm: 'volunteer form link',
        volunteerFormResults: 'volunteer form results link',
        slackChannelName: 'slack-channel-name',
        requestForm: 'request form link',
        requestFormResults: 'request form results link',
      } as Group,
    },
  })

  const { container, getByText } = renderUI('oakland')
  expect(getByText('Oakland')).toBeVisible()
  expect(getByText('the Oakland group')).toBeVisible()
  expect(useGetChapterBySlugQuery).toHaveBeenCalledWith({
    variables: { slug: 'oakland' },
  })

  expect(container).toMatchSnapshot()
})

it('shows an error message when chapter not found', () => {
  ;(useGetChapterBySlugQuery as jest.Mock).mockReturnValueOnce({
    loading: false,
    error: 'Whoops',
  })

  const { container, getByText } = renderUI('1')
  expect(getByText('Chapter "1" not found')).toBeVisible()
  expect(useGetChapterBySlugQuery).toHaveBeenCalledWith({
    variables: { slug: 'oakland' },
  })

  expect(container).toMatchSnapshot()
})

it('shows a loading message when loading', () => {
  ;(useGetChapterBySlugQuery as jest.Mock).mockReturnValueOnce({
    loading: true,
  })

  const { container, getByText } = renderUI('oakland')
  expect(getByText('Loading...')).toBeVisible()

  expect(container).toMatchSnapshot()
})
