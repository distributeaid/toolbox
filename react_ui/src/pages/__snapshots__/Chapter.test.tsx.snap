// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads the chapter details and renders the page 1`] = `
<div>
  <div
    class="w-full mx-auto bg-white overflow-hidden"
  >
    <div
      class="absolute top-0 right-0 hidden sm:block z-0"
    >
      <svg
        class="z-0"
        fill="none"
        height="698"
        viewBox="0 0 593 698"
        width="593"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class="z-0"
          d="M760 -96.0684H40.8515C18.7238 -96.0684 0 -79.57 0 -58.3579V344.673C0 538.724 179.574 697.422 400 697.422C620.426 697.422 800 539.51 800 344.673V-58.3579C800 -78.7844 782.128 -96.0684 760 -96.0684Z"
          fill="#3968DA"
          fill-opacity="0.06"
        />
      </svg>
    </div>
    <div
      class="container w-full max-w-6xl p-6 px-12 md:mx-auto overflow-hidden"
    >
      <div
        class="pb-16 sm:pb-32"
      >
        <a
          class="font-body text-xs sm:text-sm leading-3"
          href="/chapters"
        >
          <svg
            class="inline mr-2 sm:mr-4 mb-1"
            fill="none"
            height="13"
            viewBox="0 0 9 13"
            width="9"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.33792 11.4776C8.74305 11.0847 8.74306 10.4347 8.33792 10.0418L5.44758 7.23902C5.04245 6.84616 5.04245 6.19608 5.44758 5.80322L8.33792 3.00045C8.74306 2.60759 8.74306 1.95752 8.33792 1.56465L7.42079 0.675304C7.0329 0.299168 6.41638 0.299167 6.02849 0.675303L0.740359 5.80322C0.335224 6.19608 0.335223 6.84616 0.740358 7.23902L6.02849 12.3669C6.41637 12.7431 7.0329 12.7431 7.42078 12.3669L8.33792 11.4776Z"
              fill="#ED2E69"
            />
          </svg>
          chapter.backLink
        </a>
      </div>
      <h3
        class="font-heading font-normal text-xl sm:text-2xl leading-relaxed"
      >
        California

        â€”
        USA
      </h3>
      <h1
        class="font-heading font-semibold text-3xl sm:text-6xl leading-tight mb-8"
      >
        Oakland
      </h1>
      <div
        class="mb-8"
      >
        <div
          class="font-body text-gray-500 text-xs"
        >
          chapter.leadHeader
        </div>
        <div
          class="font-body text-lg font-bold"
        >
          Oaky
        </div>
      </div>
    </div>
    <div
      class="border-gray-400 border-t border-b w-full"
    >
      <div
        class="container w-full max-w-6xl p-6 px-12 md:mx-auto overflow-hidden"
      >
        <div
          class="sm:flex py-6 sm:py-12"
        >
          <p
            class="pb-4 font-body text-sm sm:text-lg sm:w-7/12 pb-12 sm:pr-20 whitespace-pre-line"
            data-testid="description"
          >
            the Oakland group
          </p>
          <div
            class="sm:w-5/12 sm:px-12 z-30"
          >
            <a
              class="inline-block mx-auto px-4 py-4 rounded font-mono font-bold text-xl sm:text-2xl text-center text-white shadow-button bg-mfd-cobalt-0 w-full mb-8"
              href="request form link"
              rel="noopener noreferrer"
              target="_blank"
            >
              chapter.requestSuppliesLink
            </a>
            <a
              class="inline-block mx-auto px-4 py-4 rounded font-mono font-bold text-xl sm:text-2xl text-center text-white shadow-button bg-mfd-pink-1 w-full mb-8"
              href="donation form link"
              rel="noopener noreferrer"
              target="_blank"
            >
              chapter.donateSuppliesLink
            </a>
            <a
              class="inline-block mx-auto px-4 py-4 rounded font-mono font-bold text-xl sm:text-2xl text-center text-white shadow-button bg-mfd-purple-2 w-full"
              href="volunteer form link"
              rel="noopener noreferrer"
              target="_blank"
            >
              chapter.volunteerLink
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="container w-full max-w-6xl p-6 px-12 md:mx-auto overflow-hidden"
    >
      <div
        class="flex flex-col justify-center py-6 pb-12 sm:py-12"
      >
        <h2
          class="text-center font-heading font-semibold text-2xl sm:text-5xl mb-6"
        >
          chapter.faqHeader
        </h2>
        <a
          class="inline-block mx-auto px-4 py-4 rounded font-mono font-bold text-xl sm:text-2xl text-center text-white shadow-button bg-mfd-cobalt-0 sm:w-auto"
          href="/faq"
        >
          chapter.faqLink
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders the document head 1`] = `
<head>
  <title>
    Donate Masks & PPE for Coronavirus in Oakland | Masks For Docs
  </title>
  <meta
    content="At Masks For Docs Oakland, we have one goal: Get protective equipment into the hands of healthcare workers as quickly as possible."
    data-react-helmet="true"
    name="description"
  />
  <meta
    content="Donate Masks & PPE for Coronavirus in Oakland | Masks For Docs"
    data-react-helmet="true"
    property="og:title"
  />
  <meta
    content="At Masks For Docs Oakland, we have one goal: Get protective equipment into the hands of healthcare workers as quickly as possible."
    data-react-helmet="true"
    property="og:description"
  />
</head>
`;

exports[`shows a loading message when loading 1`] = `
<div>
  <div
    class="w-full mx-auto bg-white overflow-hidden"
  >
    <div
      class="absolute top-0 right-0 hidden sm:block z-0"
    >
      <svg
        class="z-0"
        fill="none"
        height="698"
        viewBox="0 0 593 698"
        width="593"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class="z-0"
          d="M760 -96.0684H40.8515C18.7238 -96.0684 0 -79.57 0 -58.3579V344.673C0 538.724 179.574 697.422 400 697.422C620.426 697.422 800 539.51 800 344.673V-58.3579C800 -78.7844 782.128 -96.0684 760 -96.0684Z"
          fill="#3968DA"
          fill-opacity="0.06"
        />
      </svg>
    </div>
    <div
      class="container w-full max-w-6xl p-6 px-12 md:mx-auto overflow-hidden"
    >
      <div
        class="pb-16 sm:pb-32"
      >
        <a
          class="font-body text-xs sm:text-sm leading-3"
          href="/chapters"
        >
          <svg
            class="inline mr-2 sm:mr-4 mb-1"
            fill="none"
            height="13"
            viewBox="0 0 9 13"
            width="9"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.33792 11.4776C8.74305 11.0847 8.74306 10.4347 8.33792 10.0418L5.44758 7.23902C5.04245 6.84616 5.04245 6.19608 5.44758 5.80322L8.33792 3.00045C8.74306 2.60759 8.74306 1.95752 8.33792 1.56465L7.42079 0.675304C7.0329 0.299168 6.41638 0.299167 6.02849 0.675303L0.740359 5.80322C0.335224 6.19608 0.335223 6.84616 0.740358 7.23902L6.02849 12.3669C6.41637 12.7431 7.0329 12.7431 7.42078 12.3669L8.33792 11.4776Z"
              fill="#ED2E69"
            />
          </svg>
          chapter.backLink
        </a>
      </div>
      <div>
        Loading...
      </div>
    </div>
    <div
      class="container w-full max-w-6xl p-6 px-12 md:mx-auto overflow-hidden"
    >
      <div
        class="flex flex-col justify-center py-6 pb-12 sm:py-12"
      >
        <h2
          class="text-center font-heading font-semibold text-2xl sm:text-5xl mb-6"
        >
          chapter.faqHeader
        </h2>
        <a
          class="inline-block mx-auto px-4 py-4 rounded font-mono font-bold text-xl sm:text-2xl text-center text-white shadow-button bg-mfd-cobalt-0 sm:w-auto"
          href="/faq"
        >
          chapter.faqLink
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`shows an error message when chapter not found 1`] = `
<div>
  <div
    class="w-full mx-auto bg-white overflow-hidden"
  >
    <div
      class="absolute top-0 right-0 hidden sm:block z-0"
    >
      <svg
        class="z-0"
        fill="none"
        height="698"
        viewBox="0 0 593 698"
        width="593"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class="z-0"
          d="M760 -96.0684H40.8515C18.7238 -96.0684 0 -79.57 0 -58.3579V344.673C0 538.724 179.574 697.422 400 697.422C620.426 697.422 800 539.51 800 344.673V-58.3579C800 -78.7844 782.128 -96.0684 760 -96.0684Z"
          fill="#3968DA"
          fill-opacity="0.06"
        />
      </svg>
    </div>
    <div
      class="container w-full max-w-6xl p-6 px-12 md:mx-auto overflow-hidden"
    >
      <div
        class="pb-16 sm:pb-32"
      >
        <a
          class="font-body text-xs sm:text-sm leading-3"
          href="/chapters"
        >
          <svg
            class="inline mr-2 sm:mr-4 mb-1"
            fill="none"
            height="13"
            viewBox="0 0 9 13"
            width="9"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.33792 11.4776C8.74305 11.0847 8.74306 10.4347 8.33792 10.0418L5.44758 7.23902C5.04245 6.84616 5.04245 6.19608 5.44758 5.80322L8.33792 3.00045C8.74306 2.60759 8.74306 1.95752 8.33792 1.56465L7.42079 0.675304C7.0329 0.299168 6.41638 0.299167 6.02849 0.675303L0.740359 5.80322C0.335224 6.19608 0.335223 6.84616 0.740358 7.23902L6.02849 12.3669C6.41637 12.7431 7.0329 12.7431 7.42078 12.3669L8.33792 11.4776Z"
              fill="#ED2E69"
            />
          </svg>
          chapter.backLink
        </a>
      </div>
      <h3
        class="font-heading font-normal text-xl sm:text-2xl leading-relaxed"
      >
        Chapter "
        1
        " not found
      </h3>
    </div>
    <div
      class="container w-full max-w-6xl p-6 px-12 md:mx-auto overflow-hidden"
    >
      <div
        class="flex flex-col justify-center py-6 pb-12 sm:py-12"
      >
        <h2
          class="text-center font-heading font-semibold text-2xl sm:text-5xl mb-6"
        >
          chapter.faqHeader
        </h2>
        <a
          class="inline-block mx-auto px-4 py-4 rounded font-mono font-bold text-xl sm:text-2xl text-center text-white shadow-button bg-mfd-cobalt-0 sm:w-auto"
          href="/faq"
        >
          chapter.faqLink
        </a>
      </div>
    </div>
  </div>
</div>
`;
